aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -386,6 +386,91 @@ std::tuple<Tensor, Tensor, Tensor> _svd_helper_cuda_lib(const Tensor& self, bool
   return std::make_tuple(U_working_copy, S_working_copy, VT_working_copy);
 }
 
+
+// Todo: cusolverDnXpotrfBatched has some numerical issue and is not used here.
+//     A loop of cusolverDnXpotrf is used in case MAGMA is not linked in the pytorch build.
+//     We will switch to cusolverDnXpotrfBatched after the issue is fixed.
+//     See https://github.com/pytorch/pytorch/issues/53879.
+template<typename scalar_t>
+inline static void apply_cholesky_cusolver_potrf(Tensor& self_working_copy, bool upper, Tensor& infos) {
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+  const auto uplo = upper ? CUBLAS_FILL_MODE_UPPER : CUBLAS_FILL_MODE_LOWER;
+  const int64_t n = self_working_copy.size(-1);
+  const int64_t lda = std::max<int64_t>(1, n);
+  const int64_t batch_size = batchCount(self_working_copy);
+  const int64_t matrix_stride = matrixStride(self_working_copy);
+
+  scalar_t* self_working_copy_ptr = self_working_copy.data_ptr<scalar_t>();
+  int* infos_ptr = infos.data_ptr<int>();
+
+#ifdef USE_CUSOLVER_64_BIT
+  size_t worksize_device;
+  size_t worksize_host;
+  cusolverDnParams_t params;
+  cudaDataType datatype = at::cuda::solver::get_cusolver_datatype<scalar_t>();
+  TORCH_CUSOLVER_CHECK(cusolverDnCreateParams(&params));
+  at::cuda::solver::xpotrf_buffersize(handle, params, uplo, n, datatype, nullptr, lda, datatype, &worksize_device, &worksize_host);
+
+  // allocate workspace storage
+  auto& device_allocator = *at::cuda::getCUDADeviceAllocator();
+  auto workdata_device = device_allocator.allocate(worksize_device * batch_size);
+  void* workdata_device_ptr = workdata_device.get();
```

Comment:
Is this device synchronizing allocation? If it borrows from the allocated memory pool, would it better to be more conservative with the memory and move this workspace allocation inside the batching loop?

Reply:
I'm not sure if it is a device-sync allocation. If it is, it would be inevitable. Wouldn't one allocation better than n allocation in most cases?